python/tests/test_highlevel.py

```diff
@@ -1660,6 +1660,139 @@ def test_newick(self):
             for tree in ts.trees():
                 self.verify_newick(tree)
 
+    def test_as_dict_of_dicts(self):
+        for ts in get_example_tree_sequences():
+            tree = next(ts.trees())
+            adj_dod = tree.as_dict_of_dicts()
+            g = nx.DiGraph(adj_dod)
+
+            self.verify_nx_graph_topology(tree, g)
+            self.verify_nx_algorithm_equivalence(tree, g)
+            self.verify_nx_for_tutorial_algorithms(tree, g)
+        self.verify_nx_nearest_neighbor_search()
+
+    def verify_nx_graph_topology(self, tree, g):
+        self.assertSetEqual(set(tree.nodes()), set(g.nodes))
+
+        self.assertSetEqual(
+            set(tree.roots),
+            {n for n in g.nodes if g.in_degree(n) == 0}
+        )
+
+        self.assertSetEqual(
+            set(tree.leaves()),
+            {n for n in g.nodes if g.out_degree(n) == 0}
+        )
+
+        # test if tree has no in-degrees > 1
+        self.assertTrue(nx.is_branching(g))
+
+    def verify_nx_algorithm_equivalence(self, tree, g):
+        for root in tree.roots:
+            self.assertTrue(nx.is_directed_acyclic_graph(g))
+
+            # test descendants
+            self.assertSetEqual(
+                set(u for u in tree.nodes() if tree.is_descendant(u, root)),
+                set(nx.descendants(g, root)) | {root}
+            )
+
+            # test MRCA
+            if tree.num_nodes < 20:
+                for u, v in itertools.combinations(tree.nodes(), 2):
+                    self.assertEqual(
+                        tree.mrca(u, v),
+                        nx.lowest_common_ancestor(g, u, v) or -1
```

Comment:
I haven't seen the ``or -1`` before. What's this doing?

Reply:
The `or` operator returns the first argument if that argument is truthy. Otherwise it returns the second argument. It appears to be [standard python](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not).

This is a perl idiom that I must confess I am trying for the first time in python here. I just tried it on a whim to see if it works, and apparently it does. I agree that this usage is uncommon and am happy to replace it with something else if you prefer.

Reply:
Oh, and this works because an empty container is falsy. So I am asserting that the MRCA is `-1` if there is no lowest common ancestor.

Reply:
Can you change this to something more explicit please? IMO, code is like jokes: if you have to explain it...